File: WebAccounts/GeneralLedger_17082022/js/general-setup.js
$(document).ready(function () {
    generalSetupObject.cocd = $('#ddlCompany').val();
    generalSetupObject.do_loadlookup();
    generalSetupObject.do_loaddataedit();
    generalSetupObject.do_getUserPagepermission();

    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src = "https://api.ipify.org?format=jsonp&callback=ShowIP";
    document.getElementsByTagName("head")[0].appendChild(script);
});


var generalSetupObject = {
    cocd: '',
    ip: '',
    access: '',
    _vieweperm: false,
    _createperm: false,
    _editperm: false,
    _deleteperm: false,
    _menuid: '',
    _mainmenuid: '',

    Administrator_NoSequence: [],
    fa_AccMaster: [],

    do_loadlookup: () => {
        var _data = {};
        _data["cocd"] = generalSetupObject.cocd;

        var _passdata = {
            data: "",
        };
        _passdata.data = JSON.stringify(_data);

        $.ajax({
            type: "POST",
            async: false,
            url: "../handler/datahandler.aspx/loadAdministrator_NoSequence",
            data: JSON.stringify(_passdata),
            contentType: "application/json; charset=utf-8",
            dataType: "json",
            success: function (result) {
                if (!dochkses(result.d)) return;
                var obj = JSON.parse(`[${result.d}]`);

                for (var i = 0; i < obj.length; i++) {
                    var objnew = obj[i];
                    for (var key in objnew) {
                        var attrName = key;
                        if (attrName.toLowerCase() == "table") {
                            generalSetupObject.Administrator_NoSequence = JSON.stringify(objnew[key]);
                        }
                    }
                }

                generalSetupObject.do_render_lookup();
            },
            error: function (xhr, ajaxOptions, thrownError) {
                console.log(xhr.status + " - Error occurred");
            },
        });
    },

    do_render_lookup: () => {
        var _html = [];
        var cntrl_cbo = [];
        cntrl_cbo = $("#div_generalsetup").find("select");
        $.each(cntrl_cbo, function (key, value) {
            _html = [];
            var _data = JSON.parse(generalSetupObject.Administrator_NoSequence);
            $.each(_data, function (key, value) {
                _html.push(
                    "<option value='" + value.RowId + "'>" + value.NsDescription.replace(/[\r\n]+/gm, '') + " (" + value.NsCode.replace(/[\r\n]+/gm, '') + ")</option>"
                );
            });

            if (value.id != 'dd_RetainedEarningAcCd' && value.id != 'dd_PLAcCd' && value.id != 'dd_WHTaxRndingType' && value.id != 'dd_LCYRndingType' && value.id != 'dd_GSTRndingType' && value.id != 'dd_CustomsRndingType' && value.id != 'dd_ActionExceedingBudgetLimit' && value.id != 'dd_TaxCalculationDate' && value.id != 'dd_TaxCalculate') {
                $('#' + value.id).html(_html.join(""));
                $('#' + value.id).prepend("<option value='0' selected='selected'></option>");
            }
        });

    },

    do_getUserPagepermission: () => {
        MainObject.do_getuserpageaccess(generalSetupObject);
        generalSetupObject._vieweperm = MainObject.do_IsActionMenuPermission(generalSetupObject.access, 'GENERAL SET-UP', 'view');
        generalSetupObject._createperm = MainObject.do_IsActionMenuPermission(generalSetupObject.access, 'GENERAL SET-UP', 'create');
        generalSetupObject._editperm = MainObject.do_IsActionMenuPermission(generalSetupObject.access, 'GENERAL SET-UP', 'edit');
        generalSetupObject._deleteperm = MainObject.do_IsActionMenuPermission(generalSetupObject.access, 'GENERAL SET-UP', 'delete');
        generalSetupObject._mainmenuid = MainObject.do_IsActionMenuPermission(generalSetupObject.access, 'GENERAL SET-UP', 'menuid');

        if (!generalSetupObject._createperm[0]) {
            $('#btn_save').prop("disabled", true);
            $('#btn_save').attr('title', 'do not have permission to Add/Edit Record!!!');
        }

    },

    do_loaddataedit: () => {

        var _data = {};
        _data["cocd"] = generalSetupObject.cocd;

        var _passdata = {
            data: "",
        };
        _passdata.data = JSON.stringify(_data);

        $.ajax({
            type: "POST",
            url: "general-setup.aspx/doedit",
            data: JSON.stringify(_passdata),
            contentType: "application/json; charset=utf-8",
            dataType: "json",
            async: true,
            success: function (result) {
                if (!dochkses(result.d)) return;
                var rest = result.d;
                var obj = JSON.parse(`[${result.d}]`);
                for (var i = 0; i < obj.length; i++) {
                    var objnew = obj[i];
                    for (var key in objnew) {
                        var attrName = key;

                        if (attrName.toLowerCase() == "table") {
                            if (objnew[key].length > 0) {
                                generalSetupObject.fa_AccMaster = JSON.stringify(objnew[key]);
                                var _html = [];
                                var _data = JSON.parse(generalSetupObject.fa_AccMaster);
                                $.each(_data, function (key, value) {
                                    _html.push(
                                        "<option value='" + value.AcCd + "'>" + value.AcDesc.replace(/[\r\n]+/gm, '') + " (" + value.AcCd.replace(/[\r\n]+/gm, '') + ")</option>"
                                    );
                                });

                                $('#dd_RetainedEarningAcCd').html(_html.join(""));
                                $('#dd_RetainedEarningAcCd').prepend("<option value='0' selected='selected'></option>");

                                $('#dd_PLAcCd').html(_html.join(""));
                                $('#dd_PLAcCd').prepend("<option value='0' selected='selected'></option>");
                            }
                        }
                        if (attrName.toLowerCase() == "table1") {
                            if (objnew[key].length > 0) {

                                $('#txt_PostingFrom').val(objnew[key][0].PostingFrom);
                                $('#txt_PostingTo').val(objnew[key][0].PostingTo);
                                $('#txt_DiscTollerence').val(objnew[key][0].DiscTollerence);
                                $('#txt_DiscTollerenceAmt').val(objnew[key][0].DiscTollerenceAmt);

                                if (objnew[key][0].FinPostingOnItemRecpt == true) {
                                    $('#chk_FinPostingOnItemRecpt').prop('checked', true);
                                }
                                else {
                                    $('#chk_FinPostingOnItemRecpt').prop('checked', false);
                                }

                                if (objnew[key][0].FinPostingOnPackingSlip == true) {
                                    $('#chk_FinPostingOnPackingSlip').prop('checked', true);
                                }
                                else {
                                    $('#chk_FinPostingOnPackingSlip').prop('checked', false);
                                }

                                $('#dd_RetainedEarningAcCd').val(objnew[key][0].RetainedEarningAcCd);
                                $('#dd_PLAcCd').val(objnew[key][0].PLAcCd);

                                if (objnew[key][0].ExptdCostPosting == true) {
                                    $('#chk_ExptdCostPosting').prop('checked', true);
                                }
                                else {
                                    $('#chk_ExptdCostPosting').prop('checked', false);
                                }

                                if (objnew[key][0].ExptdRevenuePosting == true) {
                                    $('#chk_ExptdRevenuePosting').prop('checked', true);
                                }
                                else {
                                    $('#chk_ExptdRevenuePosting').prop('checked', false);
                                }

                                $('#txt_ExchangeRateServiceUrl').val(objnew[key][0].ExchangeRateServiceUrl);
                                $('#txt_WHTaxRndingPrecision').val(objnew[key][0].WHTaxRndingPrecision);
                                $('#txt_WHTaxDecPlaces').val(objnew[key][0].WHTaxDecPlaces);
                                $('#dd_WHTaxRndingType').val(objnew[key][0].WHTaxRndingType);
                                $('#txt_GSTRndingPrecision').val(objnew[key][0].GSTRndingPrecision);
                                $('#txt_GSTDecPlaces').val(objnew[key][0].GSTDecPlaces);
                                $('#dd_GSTRndingType').val(objnew[key][0].GSTRndingType);
                                $('#txt_LCYRndingPrecision').val(objnew[key][0].LCYRndingPrecision);
                                $('#txt_LCYDecPlaces').val(objnew[key][0].LCYDecPlaces);
                                $('#dd_LCYRndingType').val(objnew[key][0].LCYRndingType);
                                $('#txt_CustomsRndingPrecision').val(objnew[key][0].CustomsRndingPrecision);
                                $('#txt_CustomsDecPlaces').val(objnew[key][0].CustomsDecPlaces);
                                $('#dd_CustomsRndingType').val(objnew[key][0].CustomsRndingType);
                                $('#dd_BankAcNo').val(objnew[key][0].BankAcNo);
                                $('#dd_YearEndClosingBatchNo').val(objnew[key][0].YearEndClosingBatchNo);
                                $('#dd_ExchangeAdjBatchNo').val(objnew[key][0].ExchangeAdjBatchNo);
                                $('#dd_GLTranReversal').val(objnew[key][0].GLTranReversal);
                                $('#dd_BudgetNo').val(objnew[key][0].BudgetNo);
                                $('#dd_BudgetEntryNo').val(objnew[key][0].BudgetEntryNo);
                                $('#dd_RevisedBudgetNo').val(objnew[key][0].RevisedBudgetNo);
                                $('#dd_JournalBatchNo').val(objnew[key][0].JournalBatchNo);
                                $('#dd_TranReversalVchNo').val(objnew[key][0].TranReversalVchNo);
                                $('#dd_WHTaxPmtNo').val(objnew[key][0].WHTaxPmtNo);
                                $('#dd_WHTaxAdjNo').val(objnew[key][0].WHTaxAdjNo);
                                $('#dd_GSTPmtNo').val(objnew[key][0].GSTPmtNo);
                                $('#dd_GSTAdjNo').val(objnew[key][0].GSTAdjNo);
                                $('#dd_CustomsPmtNo').val(objnew[key][0].CustomsPmtNo);
                                $('#dd_CustomsAdjNo').val(objnew[key][0].CustomsAdjNo);
                                $('#dd_BankReconNo').val(objnew[key][0].BankReconNo);
                                $('#dd_BankStatementNo').val(objnew[key][0].BankStatementNo);

                                $('#dd_ActionExceedingBudgetLimit').val(objnew[key][0].ActionExceedingBudgetLimit);

                                if (objnew[key][0].BudgetMinusUnpostedActualTran == true) {
                                    $('#chk_BudgetMinusUnpostedActualTran').prop('checked', true);
                                }
                                else {
                                    $('#chk_BudgetMinusUnpostedActualTran').prop('checked', false);
                                }

                                $('#dd_TaxCalculationDate').val(objnew[key][0].TaxCalculationDate);
                                $('#dd_TaxCalculate').val(objnew[key][0].TaxCalculate);

                                if (objnew[key][0].IsBlock == true) {
                                    $('#chk_IsBlock').prop('checked', true);
                                }
                                else {
                                    $('#chk_IsBlock').prop('checked', false);
                                }

                            }
                        }
                    }
                }
            },
            failure: function (response) {
                alert(response.d);
                alert('Problem in retreiving items...');
            }
        });
    },

};

var savedata = function () {
    var validate = true;

    var startDate = new Date($('#txt_PostingFrom').val());
    var endDate = new Date($('#txt_PostingTo').val());

    if ($.trim($('#txt_PostingFrom').val()) == '') {
        validate = false;
        $.alertable.alert(`Posting From required.`);
        $("#txt_PostingFrom").focus();
        return false;
    }
    if ($.trim($('#txt_PostingTo').val()) == '') {
        validate = false;
        $.alertable.alert(`Posting To required.`);
        $("#txt_PostingTo").focus();
        return false;
    }

    else if ($.trim($('#txt_PostingTo').val()) != '') {
        if (endDate < startDate) {
            validate = false;
            $.alertable.alert(`Allow Posting to Should Greater Then Allow Posting From.`);
            $("#txt_PostingTo").focus();
            return false;
        }
    }

    var _data = {};
    if (validate == true) {

        _data["cocd"] = generalSetupObject.cocd;

        _data["PostingFrom"] = $('#txt_PostingFrom').val();
        _data["PostingTo"] = $('#txt_PostingTo').val();
        _data["DiscTollerence"] = $('#txt_DiscTollerence').val();
        _data["DiscTollerenceAmt"] = $('#txt_DiscTollerenceAmt').val();
        _data["FinPostingOnItemRecpt"] = $("#chk_FinPostingOnItemRecpt").is(':checked');
        _data["FinPostingOnPackingSlip"] = $("#chk_FinPostingOnPackingSlip").is(':checked');
        _data["RetainedEarningAcCd"] = $('#dd_RetainedEarningAcCd').val();
        _data["PLAcCd"] = $('#dd_PLAcCd').val();
        _data["ExptdCostPosting"] = $("#chk_ExptdCostPosting").is(':checked');
        _data["ExptdRevenuePosting"] = $("#chk_ExptdRevenuePosting").is(':checked');
        _data["ExchangeRateServiceUrl"] = $('#txt_ExchangeRateServiceUrl').val();
        _data["WHTaxRndingPrecision"] = $('#txt_WHTaxRndingPrecision').val();
        _data["WHTaxDecPlaces"] = $('#txt_WHTaxDecPlaces').val();
        _data["WHTaxRndingType"] = $('#dd_WHTaxRndingType').val();
        _data["GSTRndingPrecision"] = $('#txt_GSTRndingPrecision').val();
        _data["GSTDecPlaces"] = $('#txt_GSTDecPlaces').val();
        _data["GSTRndingType"] = $('#dd_GSTRndingType').val();
        _data["LCYRndingPrecision"] = $('#txt_LCYRndingPrecision').val();
        _data["LCYDecPlaces"] = $('#txt_LCYDecPlaces').val();
        _data["LCYRndingType"] = $('#dd_LCYRndingType').val();
        _data["CustomsRndingPrecision"] = $('#txt_CustomsRndingPrecision').val();
        _data["CustomsDecPlaces"] = $('#txt_CustomsDecPlaces').val();
        _data["CustomsRndingType"] = $('#dd_CustomsRndingType').val();
        _data["BankAcNo"] = $('#dd_BankAcNo').val();
        _data["YearEndClosingBatchNo"] = $('#dd_YearEndClosingBatchNo').val();
        _data["ExchangeAdjBatchNo"] = $('#dd_ExchangeAdjBatchNo').val();
        _data["GLTranReversal"] = $('#dd_GLTranReversal').val();
        _data["BudgetNo"] = $('#dd_BudgetNo').val();
        _data["BudgetEntryNo"] = $('#dd_BudgetEntryNo').val();
        _data["RevisedBudgetNo"] = $('#dd_RevisedBudgetNo').val();
        _data["JournalBatchNo"] = $('#dd_JournalBatchNo').val();
        _data["TranReversalVchNo"] = $('#dd_TranReversalVchNo').val();
        _data["WHTaxPmtNo"] = $('#dd_WHTaxPmtNo').val();
        _data["WHTaxAdjNo"] = $('#dd_WHTaxAdjNo').val();
        _data["GSTPmtNo"] = $('#dd_GSTPmtNo').val();
        _data["GSTAdjNo"] = $('#dd_GSTAdjNo').val();
        _data["CustomsPmtNo"] = $('#dd_CustomsPmtNo').val();
        _data["CustomsAdjNo"] = $('#dd_CustomsAdjNo').val();
        _data["BankReconNo"] = $('#dd_BankReconNo').val();
        _data["BankStatementNo"] = $('#dd_BankStatementNo').val();

        _data["ActionExceedingBudgetLimit"] = $('#dd_ActionExceedingBudgetLimit').val();
        _data["BudgetMinusUnpostedActualTran"] = $("#chk_BudgetMinusUnpostedActualTran").is(':checked');
        _data["TaxCalculationDate"] = $('#dd_TaxCalculationDate').val();
        _data["TaxCalculate"] = $('#dd_TaxCalculate').val();

        _data["isblocked"] = $("#chk_IsBlock").is(':checked');
        _data["ip"] = generalSetupObject.ip;

        var _passdata = {
            data: "",
        };
        _passdata.data = JSON.stringify(_data);
        //console.log(JSON.stringify(_passdata));

        var _url = "general-setup.aspx/doSave";
        $.ajax({
            type: "POST",
            url: _url,
            data: JSON.stringify(_passdata),
            contentType: "application/json; charset=utf-8",
            dataType: "json",
            async: true,
            success: function (result) {
                if (!dochkses(result.d)) return;
                let suc = result.d.toString().split("|~|")[0];
                let acid = result.d.toString().split("|~|")[1];
                if (suc.toLowerCase() == `true`) {
                    $.alertable.alert(`Data Saved Successfully.`);
                    generalSetupObject.do_loaddataedit();
                }
            },
            error: function (xhr, ajaxOptions, thrownError) {
                alert(xhr.status + " - Error occurred");
            },
        });

    }

};


var setTwoNumberDecimal = function (obj) {
    var obj_val = $('#' + obj.id).val();
    if (obj_val == '') obj_val = '0.00';
    $('#' + obj.id).val(parseFloat(obj_val).toFixed(2));
};

var onlyNumberKey = function (evt) {
    // Only ASCII character in that range allowed
    var ASCIICode = (evt.which) ? evt.which : evt.keyCode
    if (ASCIICode > 31 && (ASCIICode < 48 || ASCIICode > 57))
        return false;
    return true;
};

var ShowIP = function (response) {
    generalSetupObject.ip = response.ip;
};

File: WebAccounts.Buisness/GeneralLedger/GeneralSetup.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Data;
using System.Data.SqlClient;
using WebAccounts.Data;

namespace WebAccounts.Buisness
{
    public class GeneralSetup
    {
        public string _rowid = string.Empty;
        public string _mode = string.Empty;

        public string _PostingFrom = string.Empty;
        public string _PostingTo = string.Empty;
        public string _DiscTollerence = string.Empty;
        public string _DiscTollerenceAmt = string.Empty;
        public string _FinPostingOnItemRecpt = string.Empty;
        public string _FinPostingOnPackingSlip = string.Empty;
        public string _RetainedEarningAcCd = string.Empty;
        public string _PLAcCd = string.Empty;
        public string _ExptdCostPosting = string.Empty;
        public string _ExptdRevenuePosting = string.Empty;
        public string _ExchangeRateServiceUrl = string.Empty;
        public string _WHTaxRndingPrecision = string.Empty;
        public string _WHTaxDecPlaces = string.Empty;
        public string _WHTaxRndingType = string.Empty;
        public string _GSTRndingPrecision = string.Empty;
        public string _GSTDecPlaces = string.Empty;
        public string _GSTRndingType = string.Empty;
        public string _LCYRndingPrecision = string.Empty;
        public string _LCYDecPlaces = string.Empty;
        public string _LCYRndingType = string.Empty;
        public string _CustomsRndingPrecision = string.Empty;
        public string _CustomsDecPlaces = string.Empty;
        public string _CustomsRndingType = string.Empty;
        public string _BankAcNo = string.Empty;
        public string _YearEndClosingBatchNo = string.Empty;
        public string _ExchangeAdjBatchNo = string.Empty;
        public string _GLTranReversal = string.Empty;
        public string _BudgetNo = string.Empty;
        public string _BudgetEntryNo = string.Empty;
        public string _RevisedBudgetNo = string.Empty;
        public string _JournalBatchNo = string.Empty;
        public string _TranReversalVchNo = string.Empty;
        public string _WHTaxPmtNo = string.Empty;
        public string _WHTaxAdjNo = string.Empty;
        public string _GSTPmtNo = string.Empty;
        public string _GSTAdjNo = string.Empty;
        public string _CustomsPmtNo = string.Empty;
        public string _CustomsAdjNo = string.Empty;
        public string _BankReconNo = string.Empty;
        public string _BankStatementNo = string.Empty;
        public string _ActionExceedingBudgetLimit = string.Empty;
        public string _BudgetMinusUnpostedActualTran = string.Empty;
        public string _TaxCalculationDate = string.Empty;
        public string _TaxCalculate = string.Empty;

        public string _IsBlock = string.Empty;
        public string _CoCd = string.Empty;
        public string _created_by = string.Empty;
        public string _creator_MAC_add = string.Empty;

        public DataSet Operation(string str_ConnString, ref string str_catchmessage)
        {
            DataSet ds = null;
            try
            {
                List<SqlParameter> SqlRecordParams = new List<SqlParameter>();
                SqlRecordParams.Add(new SqlParameter("@mode", SqlDbType.NVarChar) { Value = this._mode });
                SqlRecordParams.Add(new SqlParameter("@PostingFrom", SqlDbType.DateTime) { Value = ((string.Empty == this._PostingFrom) ? (object)DBNull.Value : this._PostingFrom) });
                SqlRecordParams.Add(new SqlParameter("@PostingTo", SqlDbType.DateTime) { Value = ((string.Empty == this._PostingTo) ? (object)DBNull.Value : this._PostingTo) });
                SqlRecordParams.Add(new SqlParameter("@DiscTollerence", SqlDbType.Decimal) { Value = ((string.Empty == this._DiscTollerence) ? (object)DBNull.Value : Convert.ToDecimal(this._DiscTollerence)) });
                SqlRecordParams.Add(new SqlParameter("@DiscTollerenceAmt", SqlDbType.Decimal) { Value = ((string.Empty == this._DiscTollerenceAmt) ? (object)DBNull.Value : Convert.ToDecimal(this._DiscTollerenceAmt)) });
                SqlRecordParams.Add(new SqlParameter("@FinPostingOnItemRecpt", SqlDbType.Bit) { Value = ((string.Empty == this._FinPostingOnItemRecpt) ? (object)DBNull.Value : Convert.ToBoolean(this._FinPostingOnItemRecpt)) });
                SqlRecordParams.Add(new SqlParameter("@FinPostingOnPackingSlip", SqlDbType.Bit) { Value = ((string.Empty == this._FinPostingOnPackingSlip) ? (object)DBNull.Value : Convert.ToBoolean(this._FinPostingOnPackingSlip)) });
                SqlRecordParams.Add(new SqlParameter("@RetainedEarningAcCd", SqlDbType.NVarChar) { Value = this._RetainedEarningAcCd });
                SqlRecordParams.Add(new SqlParameter("@PLAcCd", SqlDbType.NVarChar) { Value = this._PLAcCd });
                SqlRecordParams.Add(new SqlParameter("@ExptdCostPosting", SqlDbType.Bit) { Value = ((string.Empty == this._ExptdCostPosting) ? (object)DBNull.Value : Convert.ToBoolean(this._ExptdCostPosting)) });
                SqlRecordParams.Add(new SqlParameter("@ExptdRevenuePosting", SqlDbType.Bit) { Value = ((string.Empty == this._ExptdRevenuePosting) ? (object)DBNull.Value : Convert.ToBoolean(this._ExptdRevenuePosting)) });
                SqlRecordParams.Add(new SqlParameter("@ExchangeRateServiceUrl", SqlDbType.VarChar) { Value = this._ExchangeRateServiceUrl });
                SqlRecordParams.Add(new SqlParameter("@WHTaxRndingPrecision", SqlDbType.Decimal) { Value = ((string.Empty == this._WHTaxRndingPrecision) ? (object)DBNull.Value : Convert.ToDecimal(this._WHTaxRndingPrecision)) });
                SqlRecordParams.Add(new SqlParameter("@WHTaxDecPlaces", SqlDbType.Int) { Value = ((string.Empty == this._WHTaxDecPlaces) ? (object)DBNull.Value : Convert.ToInt32(this._WHTaxDecPlaces)) });
                SqlRecordParams.Add(new SqlParameter("@WHTaxRndingType", SqlDbType.NVarChar) { Value = this._WHTaxRndingType });
                SqlRecordParams.Add(new SqlParameter("@GSTRndingPrecision", SqlDbType.Decimal) { Value = ((string.Empty == this._GSTRndingPrecision) ? (object)DBNull.Value : Convert.ToDecimal(this._GSTRndingPrecision)) });
                SqlRecordParams.Add(new SqlParameter("@GSTDecPlaces", SqlDbType.Int) { Value = ((string.Empty == this._GSTDecPlaces) ? (object)DBNull.Value : Convert.ToInt32(this._GSTDecPlaces)) });
                SqlRecordParams.Add(new SqlParameter("@GSTRndingType", SqlDbType.NVarChar) { Value = this._GSTRndingType });
                SqlRecordParams.Add(new SqlParameter("@LCYRndingPrecision", SqlDbType.Decimal) { Value = ((string.Empty == this._LCYRndingPrecision) ? (object)DBNull.Value : Convert.ToDecimal(this._LCYRndingPrecision)) });
                SqlRecordParams.Add(new SqlParameter("@LCYDecPlaces", SqlDbType.Int) { Value = ((string.Empty == this._LCYDecPlaces) ? (object)DBNull.Value : Convert.ToInt32(this._LCYDecPlaces)) });
                SqlRecordParams.Add(new SqlParameter("@LCYRndingType", SqlDbType.NVarChar) { Value = this._LCYRndingType });
                SqlRecordParams.Add(new SqlParameter("@CustomsRndingPrecision", SqlDbType.Decimal) { Value = ((string.Empty == this._CustomsRndingPrecision) ? (object)DBNull.Value : Convert.ToDecimal(this._CustomsRndingPrecision)) });
                SqlRecordParams.Add(new SqlParameter("@CustomsDecPlaces", SqlDbType.Int) { Value = ((string.Empty == this._CustomsDecPlaces) ? (object)DBNull.Value : Convert.ToInt32(this._CustomsDecPlaces)) });
                SqlRecordParams.Add(new SqlParameter("@CustomsRndingType", SqlDbType.NVarChar) { Value = this._CustomsRndingType });
                SqlRecordParams.Add(new SqlParameter("@BankAcNo", SqlDbType.BigInt) { Value = ((string.Empty == this._BankAcNo) ? (object)DBNull.Value : Int64.Parse(this._BankAcNo)) });
                SqlRecordParams.Add(new SqlParameter("@YearEndClosingBatchNo", SqlDbType.BigInt) { Value = ((string.Empty == this._YearEndClosingBatchNo) ? (object)DBNull.Value : Int64.Parse(this._YearEndClosingBatchNo)) });
                SqlRecordParams.Add(new SqlParameter("@ExchangeAdjBatchNo", SqlDbType.BigInt) { Value = ((string.Empty == this._ExchangeAdjBatchNo) ? (object)DBNull.Value : Int64.Parse(this._ExchangeAdjBatchNo)) });
                SqlRecordParams.Add(new SqlParameter("@GLTranReversal", SqlDbType.BigInt) { Value = ((string.Empty == this._GLTranReversal) ? (object)DBNull.Value : Int64.Parse(this._GLTranReversal)) });
                SqlRecordParams.Add(new SqlParameter("@BudgetNo", SqlDbType.BigInt) { Value = ((string.Empty == this._BudgetNo) ? (object)DBNull.Value : Int64.Parse(this._BudgetNo)) });
                SqlRecordParams.Add(new SqlParameter("@BudgetEntryNo", SqlDbType.BigInt) { Value = ((string.Empty == this._BudgetEntryNo) ? (object)DBNull.Value : Int64.Parse(this._BudgetEntryNo)) });
                SqlRecordParams.Add(new SqlParameter("@RevisedBudgetNo", SqlDbType.BigInt) { Value = ((string.Empty == this._RevisedBudgetNo) ? (object)DBNull.Value : Int64.Parse(this._RevisedBudgetNo)) });
                SqlRecordParams.Add(new SqlParameter("@JournalBatchNo", SqlDbType.BigInt) { Value = ((string.Empty == this._JournalBatchNo) ? (object)DBNull.Value : Int64.Parse(this._JournalBatchNo)) });
                SqlRecordParams.Add(new SqlParameter("@TranReversalVchNo", SqlDbType.BigInt) { Value = ((string.Empty == this._TranReversalVchNo) ? (object)DBNull.Value : Int64.Parse(this._TranReversalVchNo)) });
                SqlRecordParams.Add(new SqlParameter("@WHTaxPmtNo", SqlDbType.BigInt) { Value = ((string.Empty == this._WHTaxPmtNo) ? (object)DBNull.Value : Int64.Parse(this._WHTaxPmtNo)) });
                SqlRecordParams.Add(new SqlParameter("@WHTaxAdjNo", SqlDbType.BigInt) { Value = ((string.Empty == this._WHTaxAdjNo) ? (object)DBNull.Value : Int64.Parse(this._WHTaxAdjNo)) });
                SqlRecordParams.Add(new SqlParameter("@GSTPmtNo", SqlDbType.BigInt) { Value = ((string.Empty == this._GSTPmtNo) ? (object)DBNull.Value : Int64.Parse(this._GSTPmtNo)) });
                SqlRecordParams.Add(new SqlParameter("@GSTAdjNo", SqlDbType.BigInt) { Value = ((string.Empty == this._GSTAdjNo) ? (object)DBNull.Value : Int64.Parse(this._GSTAdjNo)) });
                SqlRecordParams.Add(new SqlParameter("@CustomsPmtNo", SqlDbType.BigInt) { Value = ((string.Empty == this._CustomsPmtNo) ? (object)DBNull.Value : Int64.Parse(this._CustomsPmtNo)) });
                SqlRecordParams.Add(new SqlParameter("@CustomsAdjNo", SqlDbType.BigInt) { Value = ((string.Empty == this._CustomsAdjNo) ? (object)DBNull.Value : Int64.Parse(this._CustomsAdjNo)) });
                SqlRecordParams.Add(new SqlParameter("@BankReconNo", SqlDbType.BigInt) { Value = ((string.Empty == this._BankReconNo) ? (object)DBNull.Value : Int64.Parse(this._BankReconNo)) });
                SqlRecordParams.Add(new SqlParameter("@BankStatementNo", SqlDbType.BigInt) { Value = ((string.Empty == this._BankStatementNo) ? (object)DBNull.Value : Int64.Parse(this._BankStatementNo)) });
                SqlRecordParams.Add(new SqlParameter("@ActionExceedingBudgetLimit", SqlDbType.Char) { Value = this._ActionExceedingBudgetLimit });
                SqlRecordParams.Add(new SqlParameter("@BudgetMinusUnpostedActualTran", SqlDbType.Bit) { Value = ((string.Empty == this._BudgetMinusUnpostedActualTran) ? (object)DBNull.Value : Convert.ToBoolean(this._BudgetMinusUnpostedActualTran)) });
                SqlRecordParams.Add(new SqlParameter("@TaxCalculationDate", SqlDbType.VarChar) { Value = this._TaxCalculationDate });
                SqlRecordParams.Add(new SqlParameter("@TaxCalculate", SqlDbType.VarChar) { Value = this._TaxCalculate });
                
                SqlRecordParams.Add(new SqlParameter("@IsBlock", SqlDbType.Bit) { Value = ((string.Empty == this._IsBlock) ? (object)DBNull.Value : Convert.ToBoolean(this._IsBlock)) });
                SqlRecordParams.Add(new SqlParameter("@CoCd", SqlDbType.VarChar) { Value = ((string.Empty == this._CoCd) ? (object)DBNull.Value : this._CoCd) });
                SqlRecordParams.Add(new SqlParameter("@created_by", SqlDbType.UniqueIdentifier) { Value = ((string.Empty == this._created_by) ? (object)DBNull.Value : (object)Guid.Parse(this._created_by)) });
                SqlRecordParams.Add(new SqlParameter("@creator_MAC_add", SqlDbType.VarChar) { Value = ((string.Empty == this._creator_MAC_add) ? (object)DBNull.Value : this._creator_MAC_add) });

                ds = DataHelper.ExecuteDataset(str_ConnString, "GeneralLedger_GeneralSetup_operation", SqlRecordParams.ToArray());
            }
            catch (Exception expErr)
            {
                ds = null;
                str_catchmessage = "[general.cs:Operation]" + expErr.Message;
            }
            finally { }
            return ds;
        }

    }
}
